### ISSUE TITLE
Training the custom dataset from pretrained weights, what model does the algorithm use as a default?

### ISSUE THREAD

**Issue by buyukkanber** [ultralytics/yolov5#7591]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hello,

I have been training my custom dataset from pretrained weight "yolov5x.pt" as instructed and recommended [here](https://github.com/ultralytics/yolov5/wiki/Train-Custom-Data#3-train) without using --cfg the model path.

`python train.py --img 800 --batch 8 --epochs 100 --data customdata.yaml --weights yolov5x.pt
`

What model does YOLOv5 algorithm use as a default, if we don't define the model? 
Is there any difference between 2 trainings below?

`python train.py --img 800 --batch 8 --epochs 100 --data customdata.yaml --weights yolov5x.pt
`
`python train.py --img 800 --batch 8 --epochs 100 --data customdata.yaml --weights yolov5x.pt --cfg models/yolov5x.yaml
`

Thanks

**Comment by MartinPedersenpp** [comment]
parser.add_argument('--data', type=str, default=ROOT / 'data/coco128.yaml', help='dataset.yaml path')

Taken from train.py line 479

Ohh and if you don't parse a weight argument(pretrained model) it will use randomly initialized weights

**Comment by glenn-jocher** [comment]
@buyukkanber see argparser for defaults. --cfg will cause a new model to be constructed from scratch. Any matching layers in --weights will be transferred.
https://github.com/ultralytics/yolov5/blob/2611477d2eb5725db84b8c971d102b91e36d27a5/train.py#L475-L519

**Comment by Vijay-P1999** [comment]
What is the difference between `--cfg model/yolov5s.yaml` and `--cfg model/yolov5l.yaml` ?
Will it increase the performance of the model and detecting accuracy if I use large `yolov5l.yaml`?

**Comment by glenn-jocher** [comment]
@Vijay-P1999 👋 Hello! Thanks for asking about selecting YOLOv5 🚀 models. 


## Model Selection

Larger models like YOLOv5x and [YOLOv5x6](https://github.com/ultralytics/yolov5/releases/tag/v5.0) will produce better results in nearly all cases, but have more parameters, require more CUDA memory to train, and are slower to run. For **mobile** deployments we recommend YOLOv5s/m, for **cloud** deployments we recommend YOLOv5l/x. See our README [table](https://github.com/ultralytics/yolov5#pretrained-checkpoints) for a full comparison of all models. 

<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>

- **Start from Pretrained weights.** Recommended for small to medium sized datasets (i.e. [VOC](https://github.com/ultralytics/yolov5/blob/master/data/VOC.yaml), [VisDrone](https://github.com/ultralytics/yolov5/blob/master/data/VisDrone.yaml), [GlobalWheat](https://github.com/ultralytics/yolov5/blob/master/data/GlobalWheat2020.yaml)). Pass the name of the model to the `--weights` argument. Models download automatically from the [latest YOLOv5 release](https://github.com/ultralytics/yolov5/releases).
```bash
python train.py --data custom.yaml --weights yolov5s.pt
                                             yolov5m.pt
                                             yolov5l.pt
                                             yolov5x.pt
                                             custom_pretrained.pt
```
- **Start from Scratch.** Recommended for large datasets (i.e. [COCO](https://github.com/ultralytics/yolov5/blob/master/data/coco.yaml), [Objects365](https://github.com/ultralytics/yolov5/blob/master/data/Objects365.yaml), [OIv6](https://storage.googleapis.com/openimages/web/index.html)). Pass the model architecture yaml you are interested in, along with an empty `--weights ''` argument:
```bash
python train.py --data custom.yaml --weights '' --cfg yolov5s.yaml
                                                      yolov5m.yaml
                                                      yolov5l.yaml
                                                      yolov5x.yaml
```
